private/src/avb_watchdog/IasSystemdWatchdogManager.cpp

```diff
@@ -144,8 +145,11 @@ IasWatchdogInterface* IasSystemdWatchdogManager::createWatchdog()
 IasWatchdogInterface* IasSystemdWatchdogManager::createWatchdog(uint32_t timeout, std::string watchdogName)
 {
   IasWatchdogInterface * watchdogInterface = nullptr;
+  int signed_timeout = timeout;
```

Comment:
This is weird... is the idea prevent a big timeout? Why not compare with threshold then? Commit message mentions "assigning negative timeout value to an unsigned int" - but that should be fixed at the caller, not here.

Reply:
the value could be converted into negative value when it's assigned to int if timeout value is big enough.